include/rsl/validators.hpp

```diff
@@ -0,0 +1,242 @@
+#pragma once
+
+#include <rsl/algorithm.hpp>
+
+#include <rclcpp/rclcpp.hpp>
+#include <tl_expected/expected.hpp>
+
+#include <fmt/ranges.h>
+#include <string>
+#include <vector>
+
+namespace rsl {
+
+/** @file */
+
+/**
+ * @brief TODO :(
+ */
+template <typename T>
+[[nodiscard]] auto unique(rclcpp::Parameter const& parameter) -> tl::expected<void, std::string> {
```

Comment:
It's weird to have functions named `unique` and `is_unique` which have such similar semantics. We should pick one name or the other and stick to it.